[Feat] :lipstick: 지역 버튼 컴포넌트 구현

soeun11 opened the pull request:
**시, **구 등에 사용할 버튼 컴포넌트를 구현했습니다.

## ✨ PR 유형

어떤 변경 사항이 있나요??

- [X] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제


## 🛠️ 작업내용
[ 작업한 내용을 작성해주세요 ( UI 구현이라면 사진도 같이 올려주시면 좋아요! ) ]
<img width="221" alt="스크린샷 2025-01-24 오전 11 43 08" src="https://github.com/user-attachments/assets/3da99975-5d8a-435d-ba49-fc6971c29fa5" />

**시, **구 등에 사용할 버튼 컴포넌트를 구현했습니다.

## 📋 추후 진행 상황
[ 다음에 진행할 작업에 대해 작성해주세요!! ]</br>
[ 현재 커밋 후 풀리퀘 다음으로 작업 내용을 적어주면 됩니다! ]

## 📌 리뷰 포인트
[ 어떤 부분을 잘 체크해야하는지 작성해주세요 ]



## ✅ Checklist

PR이 다음 요구 사항을 충족하는지 확인해주세요!!!

- [X] 커밋 메시지 컨벤션에 맞게 작성했습니다
    -   해당 링크 : [깃 모지 컨벤션](https://tngusmiso.tistory.com/57)
- [X] 유지-보수를 위해 주석처리를 잘 작성하였는가?
    - 해당 링크 : [Xcode 주석 정리](https://yoojin99.github.io/app/Swift-Documentation/)

soeun11 committed:
:lipstick: 지역 버튼 컴포넌트 구현

**시, **구 등에 사용할 버튼 컴포넌트를 구현했습니다.

JEONG-J commented:
1. UIComponents에 뷰 파일만 넣어서 해줘!
2. 지역 이름에 따라 버튼 내부의 텍스트 값이 바뀌는 거잖아?

JEONG-J commented on Catchy/Catchy/Common/UIComponents/LocationButton/LocationButtonView.swift:
```diff
@@ -0,0 +1,51 @@
+    /// 버튼의 UI를 구성하는 메서드
+    func buttonContent() -> some View {
+        Text(model.title)
+            .font(.custom("Medium", size: 13))
```
폰트 우리 폰트에서 쓸 것 ,,, 커스텀으로 만들지 말고 그 피그마에 폰트가 스타일 지정 안되어있어도 결국 지정된 스타일중에 어느 폰트를 쓴거니까 Font 파일 가서 찾아서 바꿔줘 만약 없으면 추가하고!

soeun11 replied:
넹 .. 수정하겠습니다

soeun11 commented:
네 ! 수정할게요!

soeun11 committed:
LocationButton 수정

JEONG-J reviewed (commented):
일단.. Deprecated 다시 수정해조..

.foregroundColor(isSelected ? Color.m6 : Color.g4) //글씨 색깔

.font(.system(size: 14, weight: .medium))

제발.. 피그마에 없는 폰트쓰지 말고 풀리퀘 날리기전에 꼼꼼히봐줘

뷰를 제대로 못 만드는건 익숙하지 않아서라고 이해할 수 있지만 모디피어 값을 제대로 쓰고 있는지는 본인이 코드 작성할 때 꼼꼼히 작성하고 있지 않기 때문이라고 생각이들어.. 그래서 정말 잘 작성했는지 올리기 전에 다시 보는 습관을 가지길 바라면서 수정을 부탁해~!

또 수정할 부분은 지금 이 로직이 과연 옳은걸까? 다시 생각해볼래? 버튼이 눌렸을 때 색이 변하도록 시도한건 아주 좋아 잘했어 하지만 버튼이 여러개 있는 상태에서 단 하나의 버튼만 눌려야 하는 상황으로 만드는게 우리 그룹탭에서 지역 버튼의 역할인데 이 로직에는 옳은걸까?

컴포넌트를 만드는데만 신경쓰지 말고, 컴포넌트를 만들고 사용될 뷰에서 어떻게 쓰일지도 생각해보고 고민하고 만들었으면 좋겠어!

soeun11 committed:
지역버튼 수정

LocationSelectView는 아직 완성되지않은 부분입니다! 버튼 컴포넌트를 이렇게 사용하는게 맞는지 잘 모르겠어서 확인해주시면 감사하겠습니다. 그리고 버튼도 @Binding으로 바꿔서 상위뷰에서 전달받도록 고쳤습니다.
